### DISCUSSION TITLE
html comment throws livewire off

### DISCUSSION THREAD

**Discussion by mokhosh** [livewire/livewire#3295] [7. Bugs (archived)] [answered]
If you put an html comment at the start of a livewire component it will break:
```html
<!-- some comment -->
<div class="actual component wrapper">
...
</div>
```

**Comment by joshhanley** [comment, marked as answer]
@mokhosh technically a html comment is a valid html element, and as Livewire requires one root element, I'd suggest putting the comment inside the div.

But I've worked on a PR that detects and warns about multiple root elements, and tries to detect comments to skip them when injecting Livewire. So I've suggested a potential work around in that PR https://github.com/livewire/livewire/pull/3289

Hope this helps!

**Reply by mokhosh** [reply]
Yeah i never do HTML comments.
I was using tailwind ui and they usually have comments at the beginning. So i thought i would report.

**Reply by joshhanley** [reply]
Ah yeah, makes sense. Definitely would be nice to support that
